### PR TITLE
[#4825] Log prefill retrieve empty only for the used authentication flow

### LINKED ISSUE #4825: Daily digest reporting prefill failures for forms that work with more then 1 auth "flow"
### Product versie / Product version

2.7.8 - 2.8.1

### Customer reference

SED 102

### Omschrijf het probleem / Describe the bug

Most environments use a common NAW-stap that allows for multiple authentication: DigiD + eHerkenning and anonymous. 

These steps usually consist of 2 or 3 field sets and depending on which form of authentication is used, the right field set will be shown with pre-filled values.

In a situation where a users logs into a form with DigiD a request is made to fetch user data  from the BRP, and not to the KvK to fetch the business data. 

However in the Logs we can see something is going on, but wrongly log a request has been made and the response was empty. These log items also get reported in the daily digest as failures. 

Example with eHerkenning login:
image

Example with DigiD login: 
image

Both submissions show 1 failed and 1 successful pre-fill attempt. But in both causes the failed attempt seems to be "false" (we don't see any outgoing requests, no sentry errors)

As these steps/forms are used on almost all environments, the extra "failures" on the daily digest cause confusion and hinders the visibility of actual issues.

### PR DESCRIPTION
**Pull request by vaszig**
Closes #4825

**Changes**

- Updated logging to only log empty data for the authentication flow that is used in the submission.

**Checklist**

Check off the items that are completed or not relevant.

- Impact on features

  - [x] Checked copying a form
  - [x] Checked import/export of a form
  - [x] Config checks in the configuration overview admin page
  - [x] Problem detection in the admin email digest is handled

- Release management

  - [x] I have labelled the PR as "needs-backport" accordingly

- I have updated the translations assets (you do NOT need to provide translations)

  - [x] Ran `./bin/makemessages_js.sh`
  - [x] Ran `./bin/compilemessages_js.sh`

- Dockerfile/scripts

  - [x] Updated the Dockerfile with the necessary scripts from the `./bin` folder

- Commit hygiene

  - [x] Commit messages refer to the relevant Github issue
  - [x] Commit messages explain the "why" of change, not the how

### CODE REVIEW

**Comment by vaszig** [comment]
This is a quick fix but I would like your opinion on this before I try to write some tests.

File: src/openforms/prefill/sources.py [hunk 1]
```diff
@@ -70,7 +70,12 @@ def invoke_plugin(
             if values:
                 logevent.prefill_retrieve_success(submission, plugin, fields)
             else:
-                logevent.prefill_retrieve_empty(submission, plugin, fields)
+                if (
+                    plugin.requires_auth is None
+                    or (auth_info := getattr(submission, "auth_info", None))
+                    and auth_info.attribute == plugin.requires_auth
```
Comment (sergei-maertens) [inline src/openforms/prefill/sources.py:76]:
```suggestion
                    (required_attribute := plugin.requires_auth) is None
                    or (
                        (auth_info := getattr(submission, "auth_info", None))
                        and auth_info.attribute == required_attribute
                   )
```

* the walrus operator on `plugin.requires_auth` makes it easier to interpret the last condition
* the `and` has precedence, but I prefer wrapping this for readability

**Review by sergei-maertens — APPROVED** [review]

**Comment by vaszig** [comment]
@sergei-maertens I have a difficulty backporting this because we did the refactor in v3.0 and just cherry-picking the commit will not work (the `sources.py` is a new file in 3.0).

**Comment by vaszig** [comment]
Backports:

- stable/2.8.x: 0ccae6f91e76e62a4dbc4982210fbaed235b3b9b
- stable/2.7.x: d8f0cc458a3ce55490f63fc3a665ffbcec3ba03b

**Pull request merged**